src/core/layout/qgslayoutexporter.cpp

```diff
@@ -292,6 +293,8 @@ QImage QgsLayoutExporter::renderRegionToImage( const QRectF &region, QSize image
   QImage image( QSize( width, height ), QImage::Format_ARGB32 );
   if ( !image.isNull() )
   {
+    if ( ( width * height ) > 32768 )
+      QgsMessageLog::logMessage( QObject::tr( "Error: output is bigger than 32768 pixel, result will be clipped" ) );
```

Comment:
I understand that the limitation is on width OR height, so I think this should be:
```suggestion
    // see https://doc.qt.io/qt-5/qpainter.html#limitations
    if ( width > 32768 || height > 32768 )
      QgsMessageLog::logMessage( QObject::tr( "Error: output width or height is larger than 32768 pixel, result will be clipped" ) );
```

Reply:
Yeah I wasn't too sure, but I think you're right.